docs/data/material/components/css-baseline/css-baseline.md

```diff
@@ -78,11 +78,14 @@ The colors of the scrollbars can be customized to improve the contrast (especial
 
 ```jsx
 import darkScrollbar from '@mui/material/darkScrollbar';
+
+const defaultTheme = createTheme();
+
 const theme = createTheme({
   components: {
     MuiCssBaseline: {
       styleOverrides: {
-        body: theme.palette.mode === 'dark' ? darkScrollbar() : null,
+        body: defaultTheme.palette.mode === 'dark' ? darkScrollbar() : null,
```

Comment:
@ZeeshanTamboli Can you check if this works?

```js
styleOverrides: ({ theme }) => ({
  body: theme.palette.mode === 'dark' ? darkScrollbar() : null
})
```

I think this is better if you have a custom theme.

Reply:
Does Material UI support callback in `styleOverrides` key? If yes, I think we lack documentation. I thought it was only for slots within.

----

Also the issue I linked in the description #33334 (not related to this PR fix), the user used `styleOverride` callback and accessed the theme parameter and the styles do get applied. 😓 

Here is the [CodeSandbox](https://codesandbox.io/s/pensive-varahamihira-xgryyn?file=/src/App.tsx) he provided saying there is a type error. Works in JS.

I had closed that issue understanding that we don't support callback in styleOverrides.

I am a bit confused really.

Reply:
@mnajdova I meant callback in `styleOverrides` key itself. I know it's supported for slots within styleOverrides.

Reply:
> Does Material UI support callback in styleOverrides key? If yes, I think we lack documentation. I thought it was only for slots within.

I just found this out as well and the answer is Yes, it does (CssBaseline is special because it uses `GlobalStyles`, not `styled`)  but:
- typings is incorrect
   ```ts
   // packages/mui-material/src/styles/overrides.d.ts
   export type ComponentsOverrides<Theme = unknown> = {
      [Name in keyof ComponentNameToClassKey]?: Partial<
        OverridesStyleRules<ComponentNameToClassKey[Name], Name, Theme>
      >;
    } & {
      // this should support callback as well
      MuiCssBaseline?: CSSObject | string;
    };
   ```
- yes, we lack documentation about it.
- a `.spec.tsx` would be nice

Reply:
@ZeeshanTamboli I think fixing them all together in this PR should be fine. I don't think there will be a lot of changes but I'll let you decide.

Reply:
@siriwatknp Can you point me to the implementation of the callback for the `styleOverrides` key? It would help me to figure out the typings. I could only find [this](https://github.com/mui/material-ui/blob/7cb56d2cef3f4e5e4f39214661da951c26eba9da/packages/mui-system/src/createStyled.js#L139-L145) but looks like it is for slots.

Reply:
It is in CssBaseline: https://github.com/siriwatknp/material-ui/blob/906db8e0885815cb365f0dce3dfaf9fada2dafe1/packages/mui-material/src/CssBaseline/CssBaseline.js#L47

Reply:
The `themeOverrides` is passed to `<GlobalStyles>` (emotion supports callback by default)

Reply:
Fixed. Ready for review.